### PR TITLE
[docs] Describe "Adding Custom Properties to Element State" (#749)

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**
by @VasilyStrelyaev 

/cc @kirovboris @lexkazakov @DevExpress/testcafe-docs

### CODE REVIEW

File: docs/articles/documentation/test-api/selecting-page-elements/selectors.md [hunk 1]
```diff
@@ -620,16 +613,56 @@ DOM node or the timeout exceeds.
...
+```js
+import { Selector } from 'testcafe'
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`
+
+test('Check Label HTML', async t => {
+    const label = Selector('label').addCustomDOMProperties({
+        innerHTML: el => el.innerHTML
+    })
```
Comment (churkin) [inline docs/articles/documentation/test-api/selecting-page-elements/selectors.md]: missed `;`

File: docs/articles/documentation/test-api/selecting-page-elements/selectors.md [hunk 2]
```diff
@@ -620,16 +613,56 @@ DOM node or the timeout exceeds.
...
+----------- | -------- | -----------
+`node`      | Object   | The DOM node.
+
+**Example**
+
+```js
+import { Selector } from 'testcafe'
+
+fixture `My fixture`
+    .page `https://devexpress.github.io/testcafe/example/`
```
Comment (churkin) [inline docs/articles/documentation/test-api/selecting-page-elements/selectors.md]: missed `;`

File: docs/articles/documentation/test-api/selecting-page-elements/selectors.md [hunk 3]
```diff
@@ -620,16 +613,56 @@ DOM node or the timeout exceeds.
 Note that you can additionally require that the node returned by the selector is visible.
 To do this, use the [visibilityCheck](selector-options.md#optionsvisibilitycheck) option.
 
+## Adding Custom Properties to Element State
+
+TestCafe allows you to extend [element state](#obtain-element-state) with custom properties calculated on the client side.
+
+To do this, use the selector's `addCustomDOMProperties` method.
+
+```text
```
Comment (kirovboris) [inline docs/articles/documentation/test-api/selecting-page-elements/selectors.md]: It should be javascript, not text.

File: docs/articles/documentation/test-api/selecting-page-elements/selectors.md [hunk 4]
```diff
@@ -620,16 +613,56 @@ DOM node or the timeout exceeds.
...
+Selector().addCustomDOMProperties({
+    property1: fn1,
+    property2: fn2,
+    /* ... */
+})
```
Comment (kirovboris) [inline docs/articles/documentation/test-api/selecting-page-elements/selectors.md]: missed `;`

File: docs/articles/documentation/test-api/selecting-page-elements/selectors.md [hunk 5]
```diff
@@ -620,16 +613,56 @@ DOM node or the timeout exceeds.
...
+Functions that calculate property values (`fn1`, `fn2`, ...) take the following parameters.
+
+Parameter   | Type     | Description
+----------- | -------- | -----------
+`node`      | Object   | The DOM node.
+
+**Example**
+
+```js
+import { Selector } from 'testcafe'
```
Comment (kirovboris) [inline docs/articles/documentation/test-api/selecting-page-elements/selectors.md]: missed `;`

File: docs/articles/documentation/test-api/selecting-page-elements/selectors.md (hunk 2 above)
Comment (kirovboris) [inline docs/articles/documentation/test-api/selecting-page-elements/selectors.md]: missed `;`

File: docs/articles/documentation/test-api/selecting-page-elements/selectors.md (hunk 3 above)
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/selecting-page-elements/selectors.md]: I don't think so. I don't know why but every where we use `text` in such cases:
[first](https://github.com/DevExpress/testcafe/blob/master/docs/articles/documentation/test-api/selecting-page-elements/selectors.md#creating-selectors), [second](https://github.com/DevExpress/testcafe/blob/master/docs/articles/documentation/test-api/assertions/index.md#assertion-structure)

For example in the second link:
text:
```text
await t.expect( actual ).eql( expected, message, options );
```
and than an example with `js`

Comment (kirovboris) [inline docs/articles/documentation/test-api/selecting-page-elements/selectors.md]: In these cases we write a formal description, so it should be something like:
`Selector().addCustomDOMProperties( { [...propertyHandlers] } );` or
`Selector().addCustomDOMProperties( { property1 => fn1, ... propertyN => fnN } );`
Besides, several strings of code without highlighting look awful in my opinion.

Comment (lexkazakov) [inline docs/articles/documentation/test-api/selecting-page-elements/selectors.md]: Text and the following code sample looks good for me.

Comment (VasilyStrelyaev) [inline docs/articles/documentation/test-api/selecting-page-elements/selectors.md]: This description looks pretty formal to me, it isn't an example.

**Pull request merged**